artemis-server/src/main/java/org/apache/activemq/artemis/core/config/storage/DatabaseStorageConfiguration.java

```diff
@@ -151,6 +175,33 @@ public void setDataSource(DataSource dataSource) {
       this.dataSource = dataSource;
    }
 
+   public JDBCConnectionProvider getConnectionProvider() {
+      if (connectionProvider == null) {
+         connectionProvider = new JDBCConnectionProvider(getDataSource());
+      }
+      return connectionProvider;
+   }
+   public void addDataSourceProperty(String key, String value) {
```

Comment:
why not directly calling `dataSourceProperties.put(key, value)`?
It will be up to the actual driver to validate the passed parameters, wdyt?